prov/rxm/src/rxm_ep.c

```diff
@@ -1169,25 +1187,41 @@ rxm_ep_inject_common(struct rxm_ep *rxm_ep, const void *buf, size_t len,
 		tx_buf->pkt.hdr.op = op;
 		tx_buf->pkt.hdr.flags = comp_flags;
 		memcpy(tx_buf->pkt.data, buf, tx_buf->pkt.hdr.size);
-		return rxm_ep_inject_send(rxm_ep, rxm_conn, tx_buf, pkt_size);
+		ret = rxm_ep_inject_send(rxm_ep, rxm_conn, tx_buf, pkt_size);
+		/* release allocated buffer for further reuse */
+		rxm_tx_buf_release(rxm_ep, tx_buf);
+		if (OFI_UNLIKELY(ret))
+			goto defer;
```

Comment:
`rxm_ep_defer_tx_inject_buf` at `defer` calls `rxm_ep_format_tx_res_lightweight` through `rxm_ep_defer_tx_inject_iov` again. I think the control flow is getting complicated to follow and should be refactored.

Reply:
I agree, I'll work to simplify this logic

Reply:
Thanks.